javaparser-core/src/main/java/com/github/javaparser/ast/type/ArrayType.java

```diff
@@ -0,0 +1,26 @@
+package com.github.javaparser.ast.type;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+
+import java.util.List;
+
+/**
+ * Every type followed by [] gets wrapped in an ArrayType for each [].
+ */
+public class ArrayType extends ReferenceType {
+    public ArrayType(Range range, Type type, List<AnnotationExpr> annotations) {
```

Comment:
Okay, this is going over my head: do you mean that ReferenceType can be abstract, and ClassOrInterfaceType, ArrayType and TypeParameter should extend it? This means that ReferenceType does not need its type field anymore? That actually seems to make a lot of sense.

Also: I thought that introducing ArrayType was a good thing because we could have annotations everywhere? I think I don't understand.